### PR TITLE
Port cmdLineTester tests into openj9

### PR DESCRIPTION
**Pull request by OscarQQ**
- issue: runtimes/backlog 373
- [ci skip]

 cmdLineTester_locales (disable on win, disable on OpenJ9 backlog 384)

 cmdLineTester_shareClassesStorageKey (only on zos)

 cmdLineTester_shareClassesBadStackMap

 cmdLineTester_dscr (only on aix)

 cmdLineTester_reflectCache

 cmdLineTester_bootStrapStaticVerify

 cmdLineTester_doProtectedAccessCheck

Signed-off-by: OscarQQ <Yixin.Qian@ibm.com>

### CODE REVIEW

**Comment by OscarQQ** [comment]
| Build       | Platform           | Java Version  | JDK Impl |
|:-------------:|:-------------:|:-----:|:-----------:|
| Grinder/10325     | ppc64le_linux | 8 | IBM|
| Grinder/10329    | s390x_linux      |   8 | IBM|
| Grinder/10324 | x86-64_linux      |    8 | IBM|
| Grinder/10327 | x86-64_windows      |    8 | IBM|
| Grinder/10323 | s390x_zos     |    8 | IBM|
| Grinder/10328 | ppc64_aix     |    8 | IBM|

**Comment by llxia** [comment]
What is the reason that cmdLineTester_locales is disabled on zos and only runs for IBM SDK? It runs on all platforms (except wins) internally.

**Review by renfeiw — APPROVED** [review]
LGTM

File: test/functional/cmdLineTests/shareClassTests/StorageKey/playlist.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+<playlist xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:noNamespaceSchemaLocation="../../../TKG/playlist.xsd">
+	<test>
+		<testCaseName>cmdLineTester_shareClassesStorageKey</testCaseName>
+		<variations>
+			<variation>NoOptions</variation>
+		</variations>
+		<command>$(JAVA_COMMAND) -Xdump -DUTILSJAR=$(Q)$(JVM_TEST_ROOT)$(D)functional$(D)cmdLineTests$(D)utils$(D)utils.jar$(Q) \
+        -DTESTDIR=$(JVM_TEST_ROOT) -DRESJAR=$(CMDLINETESTER_RESJAR) \
```
Comment (llxia) [inline test/functional/cmdLineTests/shareClassTests/StorageKey/playlist.xml:29]: TESTDIR and RESJAR are not used in storageKey.xml. Please check the other tests.

File: test/functional/cmdLineTests/shareClassTests/StorageKey/playlist.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        -DEXE=$(SQ)$(JAVA_COMMAND) -Xdump$(SQ) -Xint -jar $(CMDLINETESTER_JAR) \
```
Comment (llxia) [inline test/functional/cmdLineTests/shareClassTests/StorageKey/playlist.xml:30]: Please add $(JVM_OPTIONS). Please fix the other tests as well.

File: test/functional/cmdLineTests/shareClassTests/StorageKey/build.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		Build cmdLineTests locales
+	</description>
+
+	<!-- set properties for this build -->
+	<property name="DEST" value="${BUILD_ROOT}/functional/cmdLineTests/shareClassTests/StorageKey" />
+	<property name="src" location="." />
+
+	<target name="dist" description="generate the distribution">
+		<copy todir="${DEST}">
+			<fileset dir="${src}" includes="*.xml,*.jar"/>
```
Comment (llxia) [inline test/functional/cmdLineTests/shareClassTests/StorageKey/build.xml:37]: I do not think we have *.jar to copy. Please fix other tests as well.

File: test/functional/cmdLineTests/shareClassTests/StorageKey/playlist.xml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        -config $(Q)$(TEST_RESROOT)$(D)storageKey.xml$(Q) -explainExcludes -xids all,$(PLATFORM),$(VARIATION) -nonZeroExitWhenError; \
+        ${TEST_STATUS}</command>
+    <platformRequirements>os.zos</platformRequirements>
```
Comment (llxia) [inline test/functional/cmdLineTests/shareClassTests/StorageKey/playlist.xml:33]: Please fix the indentation in the PR

**Comment by llxia** [comment]
MANIFEST.MF does not need copyright

**Comment by llxia** [comment]
jenkins test extended all jdk8,jdk11

**Comment by llxia** [comment]
The above [windows failure](https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_extended.functional_x86-32_windows_Personal_testList_1/13/) is related to the machine issue.

Reran at https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_extended.functional_x86-32_windows_Personal_testList_1/14/

**Review by llxia — APPROVED** [review]
LGTM

**Pull request merged**